Add FAQ page

errhythm opened the pull request:
The current FAQ button redirects user to Notion which might disrupt their experience. Created FAQ page with markdown support to allow markdown formatting in question answers of the FAQ. 

![image](https://github.com/user-attachments/assets/3ff9f5a4-654e-4f4b-a8ed-8d21e044b771)

errhythm committed:
feat: add faq page

errhythm committed:
link the faq page in contributors page

riya09 commented on apps/jonogon-web-next/src/app/faq/page.tsx:
```diff
@@ -0,0 +1,55 @@
+                <ChevronDown className="w-5 h-5 flex-shrink-0 ml-2" />
+              )}
+            </CollapsibleTrigger>
+            <CollapsibleContent>
```
maybe we can keep the design like this, both question and answer in one box, what do you think?

<img width="631" alt="Screenshot 2024-09-13 at 11 01 06 PM" src="https://github.com/user-attachments/assets/ff5351d2-8a99-4b85-9fe2-137eaf6a5a8b">

errhythm replied:
Sure. It looks great. Thanks for reviewing as well. I still have a lot to learn.

riya09 commented on apps/jonogon-web-next/src/app/contributors/page.tsx:
```diff
@@ -125,7 +125,7 @@ export default function Contributors() {
 							<a
-								href="https://elvista.notion.site/Jonogon-962f56d9d6ea42d3839790c2146b7f6a"
+								href="/faq"
 								target="_blank"
```
I think this page don't have to open in new tab anymore

errhythm committed:
Update apps/jonogon-web-next/src/app/faq/page.tsx

Co-authored-by: Faria Begum Riya <riya09@users.noreply.github.com>

errhythm committed:
Update apps/jonogon-web-next/src/app/faq/page.tsx

Co-authored-by: Faria Begum Riya <riya09@users.noreply.github.com>

errhythm committed:
Update apps/jonogon-web-next/src/app/faq/page.tsx

Co-authored-by: Faria Begum Riya <riya09@users.noreply.github.com>

riya09 commented on apps/jonogon-web-next/src/app/faq/page.tsx:
```diff
@@ -0,0 +1,61 @@
+import { faqItems, FAQItem } from './faqData';
+
+export default function FAQ() {
+  const [openItems, setOpenItems] = useState<number[]>([]);
```
you don't need this anymore i guess

riya09 replied:
> you don't need this anymore i guess

make changes in your code, commit it and push it here.

errhythm replied:
Updated. Please check.

riya09 commented on apps/jonogon-web-next/src/app/faq/page.tsx:
```diff
@@ -0,0 +1,61 @@
+          >
+            <CollapsibleTrigger className="flex justify-between items-center w-full py-4 px-6 bg-[#F7F2EE] bg-opacity-10 text-black hover:text-red-500 font-medium rounded-md border-2 border-black border-opacity-10 transition-colors duration-200">
+              <span className="font-sans text-left">{item.question}</span>
+              {openItems[index] ?
```
```suggestion
              {openItems[index] ? (
```

riya09 commented on apps/jonogon-web-next/src/app/faq/page.tsx:
```diff
@@ -0,0 +1,61 @@
+      [index]: !prev[index]
+    }));
+  };
+    setOpenItems(prev => 
+      prev.includes(index) ? prev.filter((i: number) => i !== index) : [...prev, index]
+    );
+  };
```
remove this section
```suggestion
    
```

errhythm committed:
Update apps/jonogon-web-next/src/app/faq/page.tsx

Co-authored-by: Faria Begum Riya <riya09@users.noreply.github.com>

errhythm committed:
Update apps/jonogon-web-next/src/app/faq/page.tsx

Co-authored-by: Faria Begum Riya <riya09@users.noreply.github.com>

errhythm committed:
refactor: Update FAQ component structure suggested by @riya09

riya09 reviewed (dismissed):
Looks good 👍🏼 . Great work.

Just a small suggestion. The notion page also have other content like community guidelines, manifesto and press. So, better to add a question in faq and link the notion to the answer.

The question can be something like this-

What are the community guidelines and manifesto?

To know more about the community guidelines, manifesto and press releases [click here](https://elvista.notion.site/Jonogon-962f56d9d6ea42d3839790c2146b7f6a)

errhythm committed:
Added link to the notion page in FAQ. Fixed markdown formatting with `prose` class.

errhythm commented:
Thanks a lot for the suggestions. I have added the FAQ question. Also added `prose` class in question and answer to properly format the text so no need to include components explicitly. 

![image](https://github.com/user-attachments/assets/90197fc0-4339-4abe-aa4d-9d45a25533c0)
